File: src/view/renderer/RendererFactory.java
package view.renderer;

public class RendererFactory {
    public static Renderer getRenderer(String type) {
        RendererImplConsole rendererImplConsole = new RendererImplConsole();
        return switch (type) {
            case "console" -> rendererImplConsole;
            case "gui" -> null;
            default -> null;
        };
    }
}

File: src/view/renderer/RendererFactory.kt
package view.renderer

object RendererFactory {
    fun getRenderer(type: String): Renderer? {
        val rendererImplConsole = RendererImplConsole()
        return when (type) {
            "console" -> rendererImplConsole
            "gui" -> null
            else -> null
        }
    }
}
